op-node/eth/account_proof.go

```diff
@@ -21,24 +27,68 @@ type AccountResult struct {
 	CodeHash    common.Hash    `json:"codeHash"`
 	Nonce       hexutil.Uint64 `json:"nonce"`
 	StorageHash common.Hash    `json:"storageHash"`
-	// storageProof field is ignored, we only need to proof the account contents,
-	// we do not access any individual storage values.
+
+	// Optional
+	StorageProof []StorageProofEntry `json:"storageProof,omitempty"`
 }
 
-// Verify an account proof from the getProof RPC. See https://eips.ethereum.org/EIPS/eip-1186
+// Verify an account (and optionally storage) proof from the getProof RPC. See https://eips.ethereum.org/EIPS/eip-1186
 func (res *AccountResult) Verify(stateRoot common.Hash) error {
```

Comment:
The other approach is to use MPT prove from geth & is the approach used in the withdrawals proof package.
https://github.com/ethereum-optimism/optimism/blob/48f3f508871cc0ca20149480de1c137788f880e0/op-node/withdrawals/proof.go#L43-L100)

Reply:
Updated